File: zhihu/zhihu-backend/app/schemas/career_image.py
from __future__ import annotations

from datetime import datetime
from typing import Literal, Optional

from pydantic import BaseModel, Field


CareerImageStatus = Literal["queued", "submitted", "generating", "completed", "partial", "failed"]
VariantStatus = Literal["queued", "submitted", "generating", "completed", "failed"]


class CareerImageGenerationView(BaseModel):
    id: int
    version_number: int
    status: CareerImageStatus
    is_current: bool
    is_stale: bool
    profile_summary: dict
    style_version: str
    model: str
    landscape_size: str
    square_size: str
    landscape_status: VariantStatus
    square_status: VariantStatus
    landscape_ready: bool
    square_ready: bool
    landscape_error: Optional[str] = None
    square_error: Optional[str] = None
    submitted_at: Optional[datetime] = None
    completed_at: Optional[datetime] = None
    created_at: datetime
    updated_at: datetime


class CareerImageCurrentView(BaseModel):
    current: Optional[CareerImageGenerationView] = None
    pending: Optional[CareerImageGenerationView] = None
    can_generate: bool
    source_ready: bool
    source_message: str
    poll_interval_seconds: int = Field(3, ge=1, le=30)


class CareerImageVersionList(BaseModel):
    items: list[CareerImageGenerationView] = Field(default_factory=list)
    total: int
    page: int
    page_size: int
    total_pages: int


class CareerImageAdminItem(CareerImageGenerationView):
    user_id: int
    username: str
    provider_name: str


class CareerImageAdminList(BaseModel):
    items: list[CareerImageAdminItem]
    total: int
    page: int
    page_size: int
    total_pages: int

File: zhihu/zhihu-frontend/src/types/career-image.ts
export type CareerImageStatus = "queued" | "submitted" | "generating" | "completed" | "partial" | "failed";
export type CareerImageVariantStatus = "queued" | "submitted" | "generating" | "completed" | "failed";

export interface CareerImageGeneration {
  id: number;
  version_number: number;
  status: CareerImageStatus;
  is_current: boolean;
  is_stale: boolean;
  profile_summary: Record<string, unknown>;
  style_version: string;
  model: string;
  landscape_size: string;
  square_size: string;
  landscape_status: CareerImageVariantStatus;
  square_status: CareerImageVariantStatus;
  landscape_ready: boolean;
  square_ready: boolean;
  landscape_error: string | null;
  square_error: string | null;
  submitted_at: string | null;
  completed_at: string | null;
  created_at: string;
  updated_at: string;
}

export interface CareerImageCurrent {
  current: CareerImageGeneration | null;
  pending: CareerImageGeneration | null;
  can_generate: boolean;
  source_ready: boolean;
  source_message: string;
}

export interface CareerImageVersionList {
  items: CareerImageGeneration[];
  total: number;
  page: number;
  page_size: number;
  total_pages: number;
}

export interface CareerImageAdminItem extends CareerImageGeneration {
  user_id: number;
  username: string;
  provider_name: string;
}

export interface CareerImageAdminList {
  items: CareerImageAdminItem[];
  total: number;
  page: number;
  page_size: number;
  total_pages: number;
}
